### PR TITLE
WIP network simulator

### PR DESCRIPTION
**Pull request by hdevalence**

### CODE REVIEW

File: simulator/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+async fn main() {
+    color_backtrace::install();
+
+    let filter = EnvFilter::try_from_default_env()
+        .or_else(|_| EnvFilter::try_new("trace"))
+        .unwrap();
+
+    tracing_subscriber::fmt()
+        .with_target(false)
+        .finish()
```
Comment (hawkw) [inline simulator/src/main.rs]: try:
```suggestion
    tracing_subscriber::registry()
       .with(tracing_subscriber::fmt::layer().with_target(false))
```
here, and then filtering should behave the way you expect it to. the `fmt()` method returns a `Subscriber` that already comes with its own filtering

File: simulator/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        .with(ErrorLayer::default())
+        .with(filter)
+        .init();
```
Comment (hawkw) [inline simulator/src/main.rs]: alternatively, you could pass your filter into the subscriber builder's `with_env_filter` method, to override the default filter.
so,
```suggestion
    tracing_subscriber::fmt()
        .with_target(false)
        .with_env_filter(filter)
        .finish()
        .with(ErrorLayer::default())
        .init();
```
would also work

Comment (hdevalence) [reply simulator/src/main.rs]: this is instead of the above?

Comment (hdevalence) [reply simulator/src/main.rs]: or should i use both of these suggestions?

**Pull request merged**